packages/form-core/tests/FormApi.spec.ts

```diff
@@ -3639,3 +3639,61 @@ it('should mark sourceMap as undefined when async field error is resolved', asyn
 
   expect(field.getMeta().errorSourceMap.onChange).toBeUndefined()
 })
+
+it('should reset nested fields', () => {
+  const defaultValues = {
+    shallow: '',
+    nested: {
+      field: {
+        name: '',
+      },
+    },
+  }
+
+  const form = new FormApi({
+    defaultValues,
+  })
+  form.mount()
+
+  form.setFieldValue('shallow', 'Shallow')
+  form.setFieldValue('nested.field.name', 'Nested')
+
+  expect(form.state.values.shallow).toEqual('Shallow')
+  expect(form.state.values.nested.field.name).toEqual('Nested')
+
+  form.resetField('shallow')
+  expect(form.state.values.shallow).toEqual('')
+
+  form.resetField('nested.field.name')
+  expect(form.state.values.nested.field.name).toEqual('')
+})
+
+it('should preserve nested fields on resetField if defaultValues is not provided', () => {
+  const defaultValues = {
+    shallow: '',
+    nested: {
+      field: {
+        name: '',
+      },
+    },
+  }
+
+  const form = new FormApi({
+    defaultState: {
+      values: defaultValues,
+    },
+  })
+  form.mount()
+
+  form.setFieldValue('shallow', 'Shallow')
+  form.setFieldValue('nested.field.name', 'Nested')
+
+  expect(form.state.values.shallow).toEqual('Shallow')
+  expect(form.state.values.nested.field.name).toEqual('Nested')
+
+  form.resetField('shallow')
+  expect(form.state.values.shallow).toEqual('Shallow')
```

Comment:
I couldn't find any docs reference to my question so I'm doing it here, shouldn't `reset` without a value reset the form back to its original form's defaultValue? In this case `''`?

Reply:
@LeCarbonator was kind enough to explain to me on discord that there is no default value set on this test so the expectation is correct.